### PR TITLE
fix: LSDV-5526: Fix duplication of classifications in LLM scenario

### PR DESCRIPTION
**Pull request by Gondragos**
This PR should fix some issues in existed LLM scenario such as:
- After editing text and with a successful response, textarea doesn't get replaced (it also causes areas duplication)
- Getting errors from ml-backend breaks the current regions state

### PR fulfills these requirements
- [x] Commit message(s) and PR title follows the format `[fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made` ex. `fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors`
- [x] Tests for the changes have been added/updated (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [x] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [x] Self-reviewed and ran all changes on a local instance (for bug fixes/features)

#### Change has impacts in these area(s)
_(check all that apply)_
- [ ] Product design
- [x] Frontend

#### What does this fix?
It fixes areas duplication and stop from making any changes in region and suggestion lists if we get null instead of new suggestion list. 

#### What libraries were added/updated?
N/A

#### Does this change affect performance?
Nope

#### Does this change affect security?
Nope

#### What alternative approaches were there?
The alternative approach could be provided in the context of another tasks when we get understanding of what it should be.

#### What feature flags were used to cover this change?
`fflag_feat_all_lsdv_e_294_llm_annotations_180723_long`

### Does this PR introduce a breaking change?
_(check only one)_
- [ ] Yes, and covered entirely by feature flag(s)
- [ ] Yes, and covered partially by feature flag(s)
- [ ] No
- [x] Not sure (briefly explain the situation below)

Enabling `flag_feat_all_lsdv_e_294_llm_annotations_180723_long` could break the previous scenarios used smart tools for Image Segmentation. 

### What level of testing was included in the change?
_(check all that apply)_
- [ ] e2e
- [x] integration
- [ ] unit

### Which logical domain(s) does this change affect?
`Auto Annotating`, `TextArea`

### CODE REVIEW

File: src/stores/Annotation/Annotation.js [hunk 1]
```diff
@@ -1274,12 +1275,40 @@ export const Annotation = types
 
     acceptSuggestion(id) {
       const item = self.suggestions.get(id);
+      let itemId = id;
+      const isGlobalClassification = item.classification;
+
+      if (isFF(FF_LLM_EPIC)) {
```
Comment (hlomzik) [inline src/stores/Annotation/Annotation.js:1281]: add a comment, please, that this piece of code prevents from creating duplicated global classifications

File: tests/functional/specs/llm/accept_suggestions.cy.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    const promptTextAarea = useTextarea('&:eq(0)');
+    const answerTextArea = useTextarea('&:eq(1)');
+
+    promptTextAarea.type(`${LLM_WHAT_DO_YOU_SEE.prompt}{shift+enter}`);
+    answerTextArea.hasValue(LLM_WHAT_DO_YOU_SEE.answer);
+    promptTextAarea.clickRowEdit(1);
+    promptTextAarea.rowType(1, `{selectAll}{backspace}${LLM_WHAT_ELSE_DO_YOU_SEE.prompt}{shift+enter}`);
+    answerTextArea.hasValue(LLM_WHAT_ELSE_DO_YOU_SEE.answer);
+
+    LabelStudio.serialize().then((result) => {
```
Comment (hlomzik) [inline tests/functional/specs/llm/accept_suggestions.cy.ts:50]: should not it be async for proper test finishing detection?

File: tests/functional/specs/llm/accept_suggestions.cy.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  const prompt = area?.results?.[0]?.value?.text?.[0];
+
+  if (prompt) {
+    const suggestions = llmTextareaSuggestions[prompt];
+
+    setTimeout(() => {
+      area.store.loadSuggestions(new Promise((resolve) => setTimeout(() => {
+        resolve(suggestions);
+      }, 1000)), x => x);
+    }, 1000);
```
Comment (hlomzik) [inline tests/functional/specs/llm/accept_suggestions.cy.ts:20]: why so long? can we reduce it to make it go faster?

**Review by hlomzik — APPROVED** [review]

File: tests/functional/specs/llm/accept_suggestions.cy.ts (hunk 3 above)
Comment (Gondragos) [reply tests/functional/specs/llm/accept_suggestions.cy.ts:20]: We surely can, but I don't know if it catch the problem when it appears.

File: tests/functional/specs/llm/accept_suggestions.cy.ts (hunk 2 above)
Comment (Gondragos) [reply tests/functional/specs/llm/accept_suggestions.cy.ts:50]: It looks like it should not. It waits for resolve as I see https://docs.cypress.io/api/commands/then#Yields. And it looks like Cypress is aware of these processes and also waits for it.

File: src/stores/Annotation/Annotation.js [hunk 4]
```diff
@@ -1290,8 +1319,10 @@ export const Annotation = types
       // hack to unlock sending textarea results
       // to the ML backen every time
       // it just sets `fromSuggestion` back to `false`
-      const isTextArea = area.results.findIndex(r => r.type === 'textarea') >= 0; 
+      const isTextArea = area.results.findIndex(r => r.type === 'textarea') >= 0;
 
+      // This is temporary exception until we find the way to do it right
+      // and this was done to keep notifications on prompt editing or fixing answer from ML backend
```
[the remark this answers is not in the document]
Comment (Gondragos) [reply src/stores/Annotation/Annotation.js:1326]: We have a couple of tickets about LLM. And in one of them we could rethink this approach. But it isn't guaranteed.

[the remark this answers is not in the document]
Comment (yyassi-heartex) [reply src/stores/Annotation/Annotation.js:1326]: ok, i think thats enough to ensure it doesnt fall through the cracks - which concerned about - but if there is a plan to look at this again as part the LLM tickets we should be ok

**Review by yyassi-heartex — APPROVED** [review]

**Pull request merged**